http/longpoll/longpoll.go

```diff
@@ -0,0 +1,75 @@
+package longpoll
+
+import (
+	"context"
+	"time"
+)
+
+// LongPollFunc should return true if long polling did
+// resolve, false otherwise. If error is returned, the
+// long polling will be canceled
+type LongPollFunc func() (bool, error)
```

Comment:
Do we really need the boolean here? Does something speak against a dedicated error like `ErrDidNotResolve`

Reply:
I like to separate the resolve from the error handling. Having the call only once is no error, and I would not use the err to direct the flow here. It is basically the same reason why a map has `value, ok := foo[asd]` and not `value, err := foo[asd]`

Reply:
There is probably no reason for or against either version, except taste. I like it with the two return values.

Reply:
But with this approach the caller needs to check the bool and the err and can't just check the error if he does not event want to retry it. It's like omitting the bool on the map example. (If you don't care about existence and can also work with the default)

Reply:
But the caller is this code and this code wants to check both the error and retry case.

Reply:
I was talking about the whole approach and it seemed unnecessary to me to open the same discussion on `Until(...) (ok bool, err error)` and `LongPollUntil(...) (ok bool, err error)`...

Reply:
Oh ok, in that case:

> It's like omitting the bool on the map example.

```ok, _ := longpull.Until(...)```

Reply:
More like `_, err := longpull.Until(...)` which brings me to my point. With a map you can do `val := m[0]` as it is a builtin feature to omit the ok, which is not possible for functions and as I don't see a valid usecase for `true, error` or `false, nil` I already see that we always omit the bool. However the other comments are more important and we should not waste to much time on this discussion. If you want to have it as is, keep it.

Reply:
Please check the current MR.